application-xpoll-api/src/main/java/com/xwiki/xpoll/rest/XPollResource.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package com.xwiki.xpoll.rest;
+
+import javax.ws.rs.Encoded;
+import javax.ws.rs.PUT;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.core.Response;
+
+/**
+ * Provides the functionality of saving the user's votes and determining the winning proposal in a desired page.
+ *
+ * @version $Id$
+ * @since 2.0.5
+ */
+@Path("/xpoll/{wikiName}/spaces/{spaceName: .+}/pages/{pageName}")
```

Comment:
Why not:

    /wikis/{wikiName}/spaces/{spaceName: .+}/pages/{pageName}/xpoll

In any case, the part ``/xpoll/{wikiName}`` is not good because the pattern is ``/<collectionName>/{itemId}``, as in ``wikis/dev`` or ``spaces/Sandbox`` or ``pages/WebHome``. So if you really want to keep this URL instead of the one I proposed then it should be:

    /xpoll/wikis/{wikiName}/spaces/{spaceName: .+}/pages/{pageName}

Reply:
Initially I used your suggestion but I had some problems and changed it wondering if it clashed in some way with other URLs. I solved the problem, that was unrelated to the url, and forgot to change it back.